mp4/mediasegment.go

```diff
@@ -88,15 +94,15 @@ func (s *MediaSegment) Encode(w io.Writer) error {
 			return err
 		}
 	}
-	if len(s.Sidxs) > 0 {
-		for i := range s.Sidxs {
-			err := s.Sidxs[i].Encode(w)
-			if err != nil {
-				return err
+	for i, f := range s.Fragments {
+		if len(s.SidxsByFrag) > i && len(s.SidxsByFrag[i]) > 0 {
```

Comment:
I don't think the check `len(s.SidxsByFrag[i]) > 0` is necessary.

Also, if we can assert/maintain that `SidxByFrag` is always the same length as `Fragments` then we don't need the first `len` check either.

Reply:
The check is necessary if we don't enforce that `SidxByFrag` is the same length as `Fragments`.